### ISSUE TITLE
Difficult to read metabolic models when all small molecules are displayed

### ISSUE THREAD

**Issue by ValWood** [geneontology/wc-gocam-viz#43]
The metabolic pathways are very hard to consume when they display all of the small molecules. 
It would be better if there was a toggle to "collapse small molecules"  so that you could see only the entities.

e.g
http://noctua.geneontology.org/workbench/noctua-alliance-pathway-preview/?model_id=gomodel%3AYeastPathways_TCA-EUK-PWY

http://noctua.geneontology.org/workbench/noctua-alliance-pathway-preview/?model_id=gomodel%3A633b013300000306

**Comment by vanaukenk** [comment]
Also, we should filter out duplicates in the list on the right side of the model.

**Comment by vanaukenk** [comment]
I think display of metabolic models needs some more discussion to address:

1. Possible different view with or without the chemicals
2. If we don't display the chemicals, what causal relations would/should exist between the genes?
3. Removing duplicate small molecules from the right-hand list.

@cmungall

**Comment by ValWood** [comment]
> If we don't display the chemicals, what causal relations would/should exist between the genes?

provides input for (rather than has input/has output). In  the instances where we do  add the chemical we use both...

**Comment by vanaukenk** [comment]
> provides input for (rather than has input/has output). In the instances where we do add the chemical we use both...

Yes, sorry, that was more of a rhetorical question - this is our only relation option, but we'd need to figure out the display when that relation is not explicit in the model.

**Comment by ValWood** [comment]
But you always need one or the other? Sometimes we add both (i.e in cases where we really want to include the chemical before they can be imputed)

<img width="1256" alt="Screenshot 2024-11-20 at 21 22 08" src="https://github.com/user-attachments/assets/5587a5e3-5d74-4064-88ce-c08f20265084">

**Comment by ValWood** [comment]
And the display works:

<img width="611" alt="Screenshot 2024-11-20 at 21 25 06" src="https://github.com/user-attachments/assets/717e114f-0d31-4e1f-bc58-00a0f0bc2eb2">

**Comment by ValWood** [comment]
Ok I get it now. If only the chemical connections are present you would need to infer the gene-gene relation...

**Comment by vanaukenk** [comment]
> Ok I get it now. If only the chemical connections are present you would need to infer the gene-gene relation...

Exactly!
